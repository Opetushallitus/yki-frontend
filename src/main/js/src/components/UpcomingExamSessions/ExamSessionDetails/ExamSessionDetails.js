import React from 'react';
import PropTypes from 'prop-types';
import { withNamespaces } from 'react-i18next';
import moment from 'moment';

import classes from './ExamSessionDetails.module.css';
import {
  DATE_FORMAT,
  DATE_FORMAT_WITHOUT_YEAR,
} from '../../../common/Constants';
import { getLanguagesWithLevelDescriptions } from '../../../util/util';
import ParticipantList from './ParticipantList/ParticipantList';

export const examSessionDetails = props => {
  return (
    <div data-cy="exam-session-details">
      <h2>
        Tutkintotilaisuus:{' '}
        {getLanguagesWithLevelDescriptions([props.examSession])[0].toLowerCase()}
        {' '}
        {moment(props.examSession.session_date).format(
          DATE_FORMAT,
        )}
      </h2>
      <ParticipantList examSession={props.examSession}/>
    </div>
  );
};

examSessionDetails.propTypes = {
  examSession: PropTypes.object.isRequired,
};

export default withNamespaces()(examSessionDetails);
