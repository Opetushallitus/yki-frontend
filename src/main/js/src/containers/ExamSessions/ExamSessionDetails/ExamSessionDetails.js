import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import moment from 'moment';

import classes from './ExamSessionDetails.module.css';
import { DATE_FORMAT } from '../../../common/Constants';
import { getLanguagesWithLevelDescriptions } from '../../../util/util';
import Spinner from '../../../components/UI/Spinner/Spinner';
import ParticipantList from '../../../components/UpcomingExamSessions/ParticipantList/ParticipantList';
import ExamSessionUpdateForm from './ExamSessionUpdateForm/ExamSessionUpdateForm';
import * as actions from '../../../store/actions/index';

export class ExamSessionDetails extends Component {
  componentDidMount = () => {
    this.props.onFetchExamSessionParticipants(
      this.props.examSession.organizer_oid,
      this.props.examSession.id,
    );
  };

  render() {
    const officeHeader = () => {
      return this.props.examSession.office_oid ? (
        <h2 className={classes.ExamSessionDetailsHeader}>
          {this.props.examSession.location[0].name}
        </h2>
      ) : null;
    };

    return (
      <div data-cy="exam-session-details">
        {officeHeader()}
        <h2 className={classes.ExamSessionDetailsHeader}>
          {this.props.t('examSession')}
          {': '}
          {getLanguagesWithLevelDescriptions([
            this.props.examSession,
          ])[0].toLowerCase()}{' '}
          {moment(this.props.examSession.session_date).format(DATE_FORMAT)}
        </h2>
        <ExamSessionUpdateForm
          onSubmit={this.props.onSubmitUpdateExamSession}
          onDelete={this.props.onSubmitDeleteExamSession}
          examSession={this.props.examSession}
        />
        {this.props.loading ? (
          <Spinner />
        ) : (
          <ParticipantList
            examSession={this.props.examSession}
            participants={this.props.participants}
            onCancel={this.props.onCancelRegistration}
          />
        )}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    participants: state.exam.participants,
    loading: state.exam.loading,
    error: state.exam.error,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onFetchExamSessionParticipants: (organizerOid, examSessionId) =>
      dispatch(
        actions.fetchExamSessionParticipants(organizerOid, examSessionId),
      ),
    onCancelRegistration: (organizerOid, examSessionId, registrationId) =>
      dispatch(
        actions.cancelRegistration(organizerOid, examSessionId, registrationId),
      ),
    errorConfirmedHandler: () => dispatch(actions.examSessionFailReset()),
  };
};

ExamSessionDetails.propTypes = {
  examSession: PropTypes.object.isRequired,
  participants: PropTypes.array,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.object,
  onFetchExamSessionParticipants: PropTypes.func.isRequired,
  onCancelRegistration: PropTypes.func.isRequired,
  onSubmitUpdateExamSession: PropTypes.func.isRequired,
  onSubmitDeleteExamSession: PropTypes.func.isRequired,
  errorConfirmedHandler: PropTypes.func.isRequired,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withTranslation()(ExamSessionDetails));
