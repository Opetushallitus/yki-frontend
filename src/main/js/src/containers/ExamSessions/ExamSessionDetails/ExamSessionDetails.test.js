import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { ExamSessionDetails } from './ExamSessionDetails';

configure({ adapter: new Adapter() });

const examSession = {
  registration_end_date: '2028-12-15',
  session_date: '2028-12-30',
  participants: 2,
  max_participants: 50,
  registration_start_date: '2018-09-01',
  language_code: 'eng',
  level_code: 'KESKI',
};

describe('<ExamSessionDetails />', () => {
  it('should render exam session details', () => {
    const wrapper = shallow(
      <ExamSessionDetails
        examSession={examSession}
        participants={[]}
        loading={false}
        t={t => t}
        onFetchExamSessionParticipants={jest.fn()}
        onCancelRegistration={jest.fn()}
        errorConfirmedHandler={jest.fn()}
        onSubmitUpdateExamSession={jest.fn()}
        onSubmitDeleteExamSession={jest.fn()}
        onRelocate={jest.fn()}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
