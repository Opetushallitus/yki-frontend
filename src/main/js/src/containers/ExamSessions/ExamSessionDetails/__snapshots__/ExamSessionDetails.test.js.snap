// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExamSessionDetails /> should render exam session details 1`] = `
<div
  data-cy="exam-session-details"
>
  <h2
    className="ExamSessionDetailsHeader"
  >
    Amiedu, Strömberginkuja 3
  </h2>
  <h2
    className="ExamSessionDetailsHeader"
  >
    examSession
    : 
    englanti - 
     
    30.12.2028
  </h2>
  <withI18nextTranslation(ExamSessionUpdateForm)
    examSession={
      Object {
        "language_code": "eng",
        "level_code": "KESKI",
        "location": Array [
          Object {
            "lang": "fi",
            "name": "Amiedu, Strömberginkuja 3",
          },
        ],
        "max_participants": 50,
        "participants": 2,
        "registration_end_date": "2028-12-15",
        "registration_start_date": "2018-09-01",
        "session_date": "2028-12-30",
      }
    }
    onDelete={[MockFunction]}
    onSubmit={[MockFunction]}
  />
  <ExamSessionPostAdmission
    examSession={
      Object {
        "language_code": "eng",
        "level_code": "KESKI",
        "location": Array [
          Object {
            "lang": "fi",
            "name": "Amiedu, Strömberginkuja 3",
          },
        ],
        "max_participants": 50,
        "participants": 2,
        "registration_end_date": "2028-12-15",
        "registration_start_date": "2018-09-01",
        "session_date": "2028-12-30",
      }
    }
  />
  <withI18nextTranslation(participantList)
    examSession={
      Object {
        "language_code": "eng",
        "level_code": "KESKI",
        "location": Array [
          Object {
            "lang": "fi",
            "name": "Amiedu, Strömberginkuja 3",
          },
        ],
        "max_participants": 50,
        "participants": 2,
        "registration_end_date": "2028-12-15",
        "registration_start_date": "2018-09-01",
        "session_date": "2028-12-30",
      }
    }
    onCancel={[MockFunction]}
    onRelocate={[MockFunction]}
    participants={Array []}
  />
</div>
`;
